lib/src/lifecycle_module.dart

```diff
@@ -177,16 +185,74 @@ abstract class LifecycleModule extends SimpleModule with Disposable {
       references.add(new Reference.followsFrom(_parentContext));
     }
 
-    return tracer.startSpan('$name.$operationName', references: references)
-      ..addTags({
-        'module.instanceId': _instanceId,
-      });
+    return tracer.startSpan(
+      '$name.$operationName',
+      references: references,
+      tags: _defaultTags,
+    );
+  }
+
+  /// Creates a span with `globalTracer` from the start of [load] until now.
+  ///
+  /// This span is intended to represent the time it takes for the module to
+  /// finish asynchronously loading any necessary data and entering a state which
+  /// is ready for user interaction.
+  ///
+  /// Any [tags] or [references] specified will be added to this span.
+  @protected
+  void specifyFirstUsefulState({
+    Map<String, dynamic> tags: const {},
+    List<Reference> references: const [],
+  }) =>
+      specifyStartupTiming(
+        StartupTimingType.firstUseful,
+        tags: tags,
+        references: references,
+      );
+
+  /// Creates a span with `globalTracer` from the start of [load] until now.
+  ///
+  /// The [specifier] indicates the purpose of this span.
+  ///
+  /// Any [tags] or [references] specified will be added to this span.
+  @protected
+  void specifyStartupTiming(
+    StartupTimingType specifier, {
+    Map<String, dynamic> tags: const {},
+    List<Reference> references: const [],
+  }) {
+    // Load didn't start
+    if (_loadContext == null || _startLoadTime == null) {
+      throw new StateError(
+          'Calling `specifyStartupTiming` before calling `load()`');
+    }
+
+    final tracer = globalTracer();
+    if (tracer == null) {
+      return null;
+    }
+
+    tracer
+        .startSpan(
+          '$name.${specifier.operationName}',
+          references: [tracer.followsFrom(_loadContext)]..addAll(references),
+          startTime: _startLoadTime,
+          tags: _defaultTags..addAll(tags),
+        )
+        .finish();
+
+    _startLoadTime = null;
   }
 
   /// Name of the module for identification in exceptions and debug messages.
   // ignore: unnecessary_getters_setters
   String get name => _defaultName;
 
+  Map<String, dynamic> get _defaultTags => {
+        'span.kind': 'client',
+        'module.instanceId': _instanceId,
```

Comment:
#not I know this wasn't changed in this PR, but should this be `.instance_id`, not `.instanceId`?